src/k8s-extension/azext_k8s_extension/partner_extensions/AzureMLKubernetes.py

```diff
@@ -623,40 +624,49 @@ def _get_relay_connection_str(
+                                                                   "hybrid_connection_name": hybrid_connection_name,
+                                                                   "name": auth_rule_name})
+
     except HttpOperationError as e:
         if e.response.status_code != 404 or get_key_only:
             raise e
         # create namespace
-        relay_namespace_params = azure.mgmt.relay.models.RelayNamespace(
-            location=cluster_location, tags=resource_tag)
-
-        async_poller = relay_client.namespaces.create_or_update(
-            resource_group_name, relay_namespace_name, relay_namespace_params)
+        async_poller = NamespaceCreate(cli_ctx=cmd.cli_ctx)(command_args={"resource_group": resource_group_name,
+                                                                          "name": relay_namespace_name,
+                                                                          "location": cluster_location,
+                                                                          "tags": resource_tag})
         while True:
             async_poller.result(15)
             if async_poller.done():
                 break
 
         # create hybrid connection
-        hybrid_connection_object = relay_client.hybrid_connections.create_or_update(
-            resource_group_name, relay_namespace_name, hybrid_connection_name, requires_client_authorization=True)
-        hc_resource_id = hybrid_connection_object.id
+        hybrid_connection_object = HycoCreate(cli_ctx=cmd.cli_ctx)(command_args={"resource_group": resource_group_name,
+                                                                                 "namespace_name": relay_namespace_name,
+                                                                                 "name": hybrid_connection_name,
+                                                                                 "requires_client_authorization": True})
+        hc_resource_id = hybrid_connection_object['id']
 
         # create authorization rule
-        auth_rule_rights = [azure.mgmt.relay.models.AccessRights.manage,
-                            azure.mgmt.relay.models.AccessRights.send, azure.mgmt.relay.models.AccessRights.listen]
-        relay_client.hybrid_connections.create_or_update_authorization_rule(
-            resource_group_name, relay_namespace_name, hybrid_connection_name, auth_rule_name, rights=auth_rule_rights)
+        auth_rule_rights = ["Manage", "Send", "Listen"]
+        HycoAuthoCreate(cli_ctx=cmd.cli_ctx)(command_args={"resource_group": resource_group_name,
+                                                           "namespace_name": relay_namespace_name,
+                                                           "hybrid_connection_name": hybrid_connection_name,
+                                                           "name": hybrid_connection_name,
```

Comment:
It should be `auth_rule_name` I think
```suggestion
                                                           "name": auth_rule_name,
```